src/backend/kms/surface/mod.rs

```diff
@@ -699,24 +717,19 @@ impl SurfaceThreadState {
         gbm: GbmAllocator<DrmDeviceFd>,
         egl: EGLContext,
     ) -> Result<()> {
-        //if let Some(egl) = egl {
         let mut renderer =
             unsafe { GlowRenderer::new(egl) }.context("Failed to create renderer")?;
         init_shaders(renderer.borrow_mut()).context("Failed to initialize shaders")?;
 
         self.api.as_mut().add_node(node, gbm, renderer);
-        /*
-        } else {
-            self.software_api.as_mut().add_node(node, gbm);
-        }
-        */
 
         Ok(())
     }
 
     fn node_removed(&mut self, node: DrmNode) {
         self.api.as_mut().remove_node(&node);
-        //self.software_api.as_mut().remove_node(node);
+        // force enumeration
+        let _ = self.api.single_renderer(&self.target_node);
```

Comment:
That's ugly, but I guess it works...

If `needs_enumeration` is true, calling `api.devices()` should also force an `enumerate` call. Which I guess would be mildly better here than using `single_renderer`.